contracts/common/helpers/deployments.ts

```diff
@@ -20,5 +21,19 @@ export async function deployContractFromArtifacts(
 ) {
   const factory = new ethers.ContractFactory(abi, bytecode, wallet);
   const contract = await factory.deploy(...args);
-  return contract.waitForDeployment();
+
+  const chainId = (await wallet.provider!.getNetwork()).chainId;
+  const txReceipt = await contract.deploymentTransaction()!.wait();
```

Comment:
```suggestion
  const txReceipt = await contract.deploymentTransaction()?.wait();
```

If `contract.deploymentTransaction()` = null then:
- ! => Will throw immediately with NodeJS runtime error
- ? => We will catch this in L28-30

Reply:
? syntax is used in other instances in this PR as well